### PR TITLE
Can set an associated object to null

### PR DESCRIPTION
**Pull request by Fridus**
Ex:  
`PUT` to `localhost:9000/tasks/4` :
```
{
    "project": {"id": null }
}
```
or
```
{
    "project": null
}
```

Result
```
{
"id": 4,
"title": "Task 4",
"description": "... 4",
"deadline": null,
"createdAt": "2015-06-09T13:22:16.761Z",
"updatedAt": "2015-06-09T13:22:19.142Z",
"projectId": null,
"project": null
}
```

### CODE REVIEW

**Comment by mbroadst** [comment]
@Fridus frankly I sort of forget what the original decision was on that huge discussion with the original auto-association issue, but perhaps we also want to support "DELETE /tasks/4/project" for this as well?

**Comment by mbroadst** [comment]
it will get sort of complicated for more exotic cases though..

**Comment by Fridus** [comment]
I found this problem on the project I'm working right now (in Angular):
I submit a form for update my record and maybe change or remove its associated object.
So, the model is modified and `PUT` with `restangular`. And I would like epilogue be able to update my record based on the object received.
Ex:
```
{ id: 1, name: 'new  name' , associatedObject: { ... or null } }
```

**Comment by Fridus** [comment]
@mbroadst `DELETE /tasks/:id/project` can to be an feature in auto-associations `hasOne` and `belongsTo`.
Add to to do list ? :)
For now, there are only the Read and List on self-associations, not the Delete

**Comment by mbroadst** [comment]
@Fridus okay, it just seemed to me that this was basically saying the same thing (I now remember the issue during discussion: does DELETE actually delete the associated resource, or does it just disassociate it?)

File: tests/resource/associations.test.js [hunk 1]
```diff
@@ -396,6 +396,56 @@ describe('Resource(associations)', function() {
...
+        }, function(error, response, body) {
+          var result = _.isObject(body) ? body : JSON.parse(body);
+          expect(result.address).to.be.null;
+
+          done();
+        });
+      });
+    });
+
+    it('should set to null the associated object', function(done) {
```
Comment (mbroadst) [inline tests/resource/associations.test.js]: ```it('should allow updating an association with null', function(done) { ```

File: tests/resource/associations.test.js [hunk 2]
```diff
@@ -396,6 +396,56 @@ describe('Resource(associations)', function() {
       });
     });
 
+    it('should set to null the associated object by identifier of object nested', function(done) {
```
Comment (mbroadst) [inline tests/resource/associations.test.js]: ```it('should allow updating an association to null by replacing its primary key', function(done) { ```  maybe?

Sorry to be a stickler about language, but I think it's important that people be able to read this very clearly in the future

Comment (Fridus) [inline tests/resource/associations.test.js]: No problem ;)

**Comment by Fridus** [comment]
@mbroadst This modification just enable to dissociate objects.

**Pull request merged**